x/mongo/driver/operation.go

```diff
@@ -482,6 +482,19 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 					ConnectionDescription: desc.Server,
 					CurrentIndex:          currIndex,
 				}
+
+				// Update HelloOK on server description if necessary.
+				//
+				// If HelloOK is still false and command sent was a legacy hello,
+				// check if server returned "helloOk", and if so, set desc.HelloOk.
+				if !desc.HelloOK && strings.ToLower(op.cmdName) == "ismaster" {
+					val, err := res.LookupErr("helloOk")
+					if err == nil {
+						if helloOK, ok := val.BooleanOK(); ok {
+							desc.HelloOK = helloOK
+						}
+					}
+				}
```

Comment:
I could not do this processing within `ProcessResponseFn` for `IsMaster` because that function doesn't have access to the server description `desc`. I thought this was the best place to look through the response. 

This logic _will_ parse every `ismaster` or `isMaster` response as long as the server does not eventually return `helloOk: true`. I'm not sure if that will be a performance hit in monitoring, but I also don't know how else to check for `helloOk: true`.

Reply:
If the state of `helloOk` is stored in `description.Server`, could parsing occur in `description.NewServer()`?

Reply:
Yep! Great idea; parsing in `description.NewServer()` makes more sense to me logically. Moved.